blaze/expr/collections.py

```diff
@@ -418,6 +418,7 @@ def on_right(self):
         return on_right
 
     @property
+    @memoize
     def schema(self):
```

Comment:
I think that you are correct about the slots not being used. The object was already weakreferencable from the super class's `__dict__`. We can probably just remove the slots or add slots to the super class.